linodecli/configuration.py

```diff
@@ -129,7 +133,94 @@ def set_default_user(self, username):
             sys.exit(1)
 
         self.config.set('DEFAULT', 'default-user', username)
-        self._write_config()
+        self.write_config()
+
+    # plugin methods - these are intended for plugins to utilize to store their
+    # own persistent config information
+    def get_value(self, key):
+        """
+        Retrieves and returns an existing config value for the current user.  This
+        is intended for plugins to use instead of having to deal with figuring out
+        who the current user is when accessing their config.
+
+        .. warning::
+           Plugins _MUST NOT_ set values for the user's config except through
+           ``plugin_set_value`` below.
+
+        :param key: The key to look up.
+        :type key: str
+
+        :returns: The value for that key, or None if the key doesn't exist for the
+                  current user.
+        :rtype: any
+        """
+        username = self.username or self.default_username()
+
+        if not self.config.has_option(username, key):
+            return None
+
+        return self.config.get(username, key)
+
+    def plugin_set_value(self, plugin_name, key, value):
```

Comment:
I think having name as a parameter here leaves too much up to the plugin. Why not have plugins register their name in a global on invoke and then read that here.

What do you think?